lib/chai/core/assertions.js

```diff
@@ -92,18 +117,19 @@ module.exports = function (chai, _) {
   /**
    * ### .nested
    *
-   * Sets the `nested` flag, later used by the `property` assertion.
+   * Enables dot- and bracket-notation in all `.property` and `.include`
+   * assertions that follow in the chain.
    *
-   *     expect({ foo: { bar: { baz: 'quux' } } })
-   *       .to.have.nested.property('foo.bar.baz', 'quux');
+   *     expect({a: {b: ['x', 'y']}}).to.have.nested.property('a.b[1]');
+   *     expect({a: {b: ['x', 'y']}}).to.nested.include({'a.b[1]': 'y'});
    *
-   * `.nested.property` special characters can be escaped by adding two slashes
-   * before the `.` or `[]`.
+   * If `.` or `[]` are part of an actual property name, they can be escaped by
+   * adding two backslashes before them.
```

Comment:
Good point! I vote to keep as-is for now to be beginner friendly, but it might be interesting in the future to add another example using the single-slash alternative.